fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/connecting/ConnectingActivity.java

```diff
@@ -0,0 +1,111 @@
+package com.github.dedis.popstellar.ui.home.connecting;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+
+import androidx.annotation.IdRes;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.github.dedis.popstellar.R;
+import com.github.dedis.popstellar.ui.detail.LaoDetailActivity;
+import com.github.dedis.popstellar.ui.home.HomeActivity;
+import com.github.dedis.popstellar.utility.ActivityUtils;
+
+import java.util.function.Supplier;
+
+import dagger.hilt.android.AndroidEntryPoint;
+
+@AndroidEntryPoint
+public class ConnectingActivity extends AppCompatActivity {
+  public static final String TAG = ConnectingActivity.class.getSimpleName();
+
+  private ConnectingViewModel mViewModel;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.connecting_activity);
+    setConnectingFragment();
+
+    String channelId = (String) (getIntent().getExtras()).get("LAO_ID");
```

Comment:
Unnecessary parenthesis

```suggestion
    String channelId = (String) getIntent().getExtras().get("LAO_ID");
```